storagemarket/provider.go

```diff
@@ -321,6 +325,9 @@ func (p *Provider) Start() ([]*dealHandler, error) {
 	}
 	log.Infow("db initialized")
 
+	// start the index provider
+	p.ip.Start()
+
```

Comment:
Why are we starting the index provider from inside the provider?

Reply:
It's a bit weird because we use the index provider externally to the storage market provider (eg to announce all deals), but we start the index provider inside the storage market provider.

I'd suggest that we call index provider start at the modules layer, maybe right after we call storage market provider start.

Reply:
Done.